ethpm/utils/chains.py

```diff
@@ -0,0 +1,105 @@
+import re
+
+from urllib import parse
+from cytoolz import curry
+
+from eth_utils import (
+    add_0x_prefix,
+    remove_0x_prefix,
+    is_integer,
+    encode_hex,
+)
+
+
+def get_chain_id(web3):
+    return web3.eth.getBlock(0)['hash']
+
+
+BLOCK = "block"
+
+BIP122_URL_REGEX = (
+    "^"
+    "blockchain://"
+    "(?P<chain_id>[a-zA-Z0-9]{64})"
+    "/"
+    "(?P<resource_type>block|transaction)"
+    "/"
+    "(?P<resource_hash>[a-zA-Z0-9]{64})"
+    "$"
+)
+
+
+def is_BIP122_uri(value):
+    return bool(re.match(BIP122_URL_REGEX, value))
+
+
+def parse_BIP122_uri(blockchain_uri):
+    match = re.match(BIP122_URL_REGEX, blockchain_uri)
+    if match is None:
+        raise ValueError("Invalid URI format: '{0}'".format(blockchain_uri))
+    chain_id, resource_type, resource_hash = match.groups()
+    return (
+        add_0x_prefix(chain_id),
+        resource_type,
+        add_0x_prefix(resource_hash),
+    )
+
+
+def is_BIP122_block_uri(value):
+    if not is_BIP122_uri(value):
+        return False
+    _, resource_type, _ = parse_BIP122_uri(value)
+    return resource_type == BLOCK
+
+
+@curry
+def check_if_chain_matches_chain_uri(web3, blockchain_uri):
+    chain_id, resource_type, resource_hash = parse_BIP122_uri(blockchain_uri)
+    genesis_block = web3.eth.getBlock('earliest')
+
+    if encode_hex(genesis_block['hash']) != chain_id:
```

Comment:
I added `encode_hex()` here and on line 68 to get it to work. Is that fine? When "stealing" utils from other repos, is it better to leave them as is - or adjust as necessary?

Reply:
Adjust as necessary.  Most of the utils you're stealing from populus are going to be removed as this library will replace them.